src/main/arm/createUiDefinition.json

```diff
@@ -181,12 +181,22 @@
                                 "Standard_B1ls",
                                 "Standard_A0",
                                 "Basic_A0",
-                                "Standard_B1s"
+                                "Standard_B1s",
+                                ${azure.armBased.vmSize.list}
                             ]
                         },
                         "osPlatform": "Linux",
                         "count": "1"
                     },
+                    {
+                        "name": "invalidVMSizeInfo",
+                            "type": "Microsoft.Common.InfoBox",
+                            "visible": "[contains(steps('ClusterConfig').vmSizeSelect,'p')]",
+                            "options": {
+                                "icon": "Error",
+                                "text": "The VM size you selected includes the feature letter 'p', indicating it uses ARM CPUs. ARM platform is not supported. Please select a different VM size. For more information, refer to the <a href='https://learn.microsoft.com/azure/virtual-machines/vm-naming-conventions' target='_blank'>Azure virtual machine sizes naming conventions</a>."
+                        }
+                    },
```

Comment:
Are you certain that any occurrence of the letter `p` in the string is a sufficient indicator? Having read the document you referenced, that one only talks about the letter `p` being in the "features" section of the string. Are you certain that no other part of the string will include a `p`?

Reply:
I've checked the vm sizes from `Microsoft.Compute.SizeSelector` and I haven't found any vm size string includes a `p` except ARM CPU based vm sizes.